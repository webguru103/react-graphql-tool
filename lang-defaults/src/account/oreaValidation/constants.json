[
  {
    "id": "cannot-reach-title",
    "defaultMessage": "We couldn't reach OREA"
  },
  {
    "id": "make-request-msg",
    "defaultMessage": "In order to continue,\n    we need to verify your identity with OREA.\n    Please click the button below to verify\n    your identity."
  },
  {
    "id": "first-login-msg-accent",
    "defaultMessage": "In order to use OREA forms,"
  },
  {
    "id": "first-login-msg",
    "defaultMessage": "you need to verify your identity with OREA.\n    Please click the button below to verify\n    your identity."
  },
  {
    "id": "make-request-title",
    "defaultMessage": "Identity verification"
  },
  {
    "id": "cannot-reach-msg",
    "defaultMessage": "Sorry, but we were not able to reach\n    OREA at this moment. You can go back and\n    try again, or verify later in your account settings."
  },
  {
    "id": "do-it-later",
    "defaultMessage": "Do It Later"
  },
  {
    "id": "try-again",
    "defaultMessage": "Try Again"
  },
  {
    "id": "do-it-later-msg",
    "defaultMessage": "You can validate with OREA anytime in the Account Settings page."
  },
  {
    "id": "ok-button",
    "defaultMessage": "OK"
  },
  {
    "id": "cancel-button",
    "defaultMessage": "Cancel"
  },
  {
    "id": "do-it-later-button",
    "defaultMessage": "I'll do it later."
  },
  {
    "id": "go-to-orea",
    "defaultMessage": "Go to OREA validation page"
  },
  {
    "id": "ver-failure-msg",
    "defaultMessage": "Sorry, but we were not able to verify your identity with OREA. Please check your information and try again."
  },
  {
    "id": "ver-failure-title",
    "defaultMessage": "Verification failed"
  },
  {
    "id": "ver-sucess-msg",
    "defaultMessage": "Your identity has been successfully verified through OREA."
  },
  {
    "id": "ver-sucess-title",
    "defaultMessage": "OREA verificiation successful"
  },
  {
    "id": "done-button",
    "defaultMessage": "Done"
  }
]