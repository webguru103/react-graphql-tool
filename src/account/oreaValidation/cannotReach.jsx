import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import Button from '../../shared/button/button';
import styles from './cannotReach.styles';
import c from './constants';
import IllSmallLock from '../shared/icons/ill_small_lock.svg';

type PropType = {
  classes: Object,
  changeView: Function,
}

export const CannotReachC = ({ classes, changeView }: PropType) => (
  <React.Fragment>
    <p className={classes.screenTitle} data-testid="cannot-reach">
      {c.CANNOT_REACH_TITLE}
    </p>
    <div className={classes.notice}>
      <div className={classes.noticeImageContainer}>
        <IllSmallLock style={{ width: '80px', transform: 'translate(-18px, -12px)' }} />
      </div>
      <div className={classes.noticeText}>
        {c.CANNOT_REACH_MSG}
      </div>
    </div>
    <div className={classes.controlButtons}>
      <Button
        testId="change-view-do-it-later"
        classes={{ button: classes.controlButton }}
        text={c.DO_IT_LATER}
        onClick={() => changeView('do-it-later')}
        secondary
      />
      <Button
        testId="change-view-make-request"
        classes={{ button: classes.controlButton }}
        text={c.TRY_AGAIN}
        onClick={() => changeView('make-request')}
      />
    </div>
  </React.Fragment>
);

export default withStyles(styles)(CannotReachC);
