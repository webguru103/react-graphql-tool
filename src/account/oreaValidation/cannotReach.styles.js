import sharedCss from '../shared/shared.styles';

export default (theme: Object) => ({
  screenContainer: sharedCss.screenContainer,
  screenTitle: sharedCss.screenTitle(theme),
  button: {
    textTransform: 'none',
    width: '100%',
    marginTop: 30,
    fontSize: 16,
    fontWeight: 300,
    height: 38,
  },
  controlButtons: {
    ...sharedCss.controlButtons,
  },
  controlButton: sharedCss.controlButton,
  notice: {
    display: 'flex',
    flexFlow: 'row nowrap',
  },
  noticeText: {
    fontSize: 16,
    color: theme.palette.neutrals.mainText,
  },
  noticeImageContainer: {
    minWidth: 70,
    minHeight: 70,
  },
  noticeImage: {
    width: '120%',
    height: '120%',
    transform: 'translate(-18px, -12px)',
  },
});
