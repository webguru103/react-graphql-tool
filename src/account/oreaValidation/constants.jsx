import React from 'react';
import { FormattedMessage } from 'react-intl';

const DEFAULT_PAGE_SIZE = 10;

const ENVIRONMENTS = {
  DEV: 'DEV',
  SIT: 'SIT',
  QA: 'QA',
  PROD: 'PROD',
};

export default {
  DEFAULT_PAGE_SIZE,
  ENVIRONMENTS,
  CANNOT_REACH_TITLE: <FormattedMessage id="cannot-reach-title" defaultMessage="We couldn&#39;t reach OREA" />,
  MAKE_REQUEST_MSG: <FormattedMessage
    id="make-request-msg"
    defaultMessage="In order to continue,
    we need to verify your identity with OREA.
    Please click the button below to verify
    your identity."
  />,
  FIRST_LOGIN_MSG_ACCENT: <FormattedMessage
    id="first-login-msg-accent"
    defaultMessage="In order to use OREA forms, "
  />,
  FIRST_LOGIN_MSG: <FormattedMessage
    id="first-login-msg"
    defaultMessage="you need to verify your identity with OREA.
    Please click the button below to verify
    your identity."
  />,
  MAKE_REQUEST_TITLE: <FormattedMessage id="make-request-title" defaultMessage="Identity verification" />,
  CANNOT_REACH_MSG: <FormattedMessage
    id="cannot-reach-msg"
    defaultMessage="Sorry, but we were not able to reach
    OREA at this moment. You can go back and
    try again, or verify later in your account settings."
  />,
  DO_IT_LATER: <FormattedMessage id="do-it-later" defaultMessage="Do It Later" />,
  TRY_AGAIN: <FormattedMessage id="try-again" defaultMessage="Try Again" />,
  DO_IT_LATER_MSG: <FormattedMessage id="do-it-later-msg" defaultMessage="You can validate with OREA anytime in the Account Settings page." />,
  OK: <FormattedMessage id="ok-button" defaultMessage="OK" />,
  CANCEL: <FormattedMessage id="cancel-button" defaultMessage="Cancel" />,
  ILL_DO_IT_LATER: <FormattedMessage id="do-it-later-button" defaultMessage="I'll do it later." />,
  GO_TO_OREA: <FormattedMessage id="go-to-orea" defaultMessage="Go to OREA validation page" />,
  VER_FAILURE_MSG: <FormattedMessage
    id="ver-failure-msg"
    defaultMessage="Sorry, but we were not able to verify your identity with OREA. Please check your information and try again."
  />,
  VER_FAILURE_TITLE: <FormattedMessage id="ver-failure-title" defaultMessage="Verification failed" />,
  VER_SUCCESS_MSG: <FormattedMessage id="ver-sucess-msg" defaultMessage="Your identity has been successfully verified through OREA." />,
  VER_SUCCESS_TITLE: <FormattedMessage id="ver-sucess-title" defaultMessage="OREA verificiation successful" />,
  DONE: <FormattedMessage id="done-button" defaultMessage="Done" />,
};
