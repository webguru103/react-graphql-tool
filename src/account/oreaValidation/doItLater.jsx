import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import Button from '../../shared/button/button';
import styles from './doItLater.styles';
import c from './constants';

type PropType = {
  classes: Object,
  closeDialog: Function,
}

export const DoItLaterC = ({ classes, closeDialog }: PropType) => (
  <React.Fragment>
    <div className={classes.notice} data-testid="do-it-later">
      <div className={classes.noticeText}>
        {c.DO_IT_LATER_MSG}
      </div>
    </div>
    <div className={classes.controlButtons}>
      <Button
        testId="close-dialog"
        classes={{ button: classes.controlButton }}
        text={c.OK}
        onClick={closeDialog}
        secondary
      />
    </div>
  </React.Fragment>
);

export default withStyles(styles)(DoItLaterC);
