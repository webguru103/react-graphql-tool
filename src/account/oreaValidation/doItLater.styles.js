import sharedCss from '../shared/shared.styles';

export default (theme: Object) => ({
  noticeText: sharedCss.noticeText(theme),
  controlButtons: sharedCss.controlButtons,
  controlButton: sharedCss.controlButton,
  notice: {
    ...sharedCss.notice,
    marginTop: 26,
  },
});
