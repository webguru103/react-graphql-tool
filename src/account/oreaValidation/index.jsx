import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import type { Match } from 'react-router';
import uuidv1 from 'uuid/v1';
import { compose, pro } from '../../utility';
import styles from './index.styles';
import MakeRequest from './makeRequest';
import CannotReach from './cannotReach';
import DoItLater from './doItLater';
import VerificationFailed from './verificationFailed';
import VerificationSuccess from './verificationSuccess';
import { withSendOREAGuid } from '../api/accountService';
import { withAppUser } from '../../shared/authorization/userConsumer';
import { OREA_VALIDATION_ENDPOINT } from '../../configurations';
import type { UserType } from '../../flowTypes';

type PropType = {
  classes: Object,
  closeDialog: Function,
  match: Match,
  initialDialog?: string,
  sendOreaGuid: Function,
  user: UserType,
  firstLogin: boolean,
}

type StateType = {
  view: string,
}

export class OREAValidationC extends React.Component<PropType, StateType> {

  static defaultProps = {
    initialDialog: 'make-request',
  };

  state = {
    view: this.props.initialDialog ? this.props.initialDialog : 'make-request',
  };

  fetchOREA = () => {
    // Test if page is live before sending POST request.
    fetch('https://www.orea.com/member-validation', { mode: 'no-cors' })
      .then(() => {
        // If page responds, assume page is live.
        // No information is returned with no-cors request to OREA.
        this.sendGuid();
      })
      .catch(() => {
        // If page cannot be fetched, send to 'cannot reach OREA' page.
        this.changeView('cannot-reach');
      });
  };

  sendGuid = async () => {
    // Send GUID to user object in our DB. Only current user has permissions to store it there.
    // Then the guid will be sent to OREA in oreaRequest POST form request, then sent back from OREA to orion.
    // When it comes back from OREA along with status of OREA membership, we can compare it to the one stored on user object.
    const guid = uuidv1();
    // TODO: Placeholder ID used - update with current user's ID (and also validate access to updateUser with JWT)
    const [err, res] = await pro(this.props.sendOreaGuid({ id: this.props.user.id, guid }));

    if (err || !res) {
      this.changeView('cannot-reach');
      return;
    }

    // Make the OREA POST request with stored guid and user ID
    this.oreaRequest(res.data.updateUser.id, guid);
  };

  testOREADown = () => {
    this.changeView('cannot-reach');
  };

  oreaRequest = (userId: string, guid: string) => {
    const { match } = this.props;

    const formData = {
      Guid: guid,
      ReturnURL: `${OREA_VALIDATION_ENDPOINT}?redirectPath=${match.path}&userId=${userId}`,
      SecurityKey: 'orea#4498',
      PartnerName: 'DealTap',
    };

    const form = document.createElement('form');
    form.style.visibility = 'hidden';
    form.method = 'post';
    form.action = 'https://www.orea.com/member-validation';
    Object.keys(formData).forEach((item) => {
      const input = document.createElement('input');
      input.style.visibility = 'hidden';
      input.name = item;
      input.value = formData[item];
      form.appendChild(input);
    });
    if (document.body) {
      document.body.appendChild(form);
    }
    form.submit();
  };

  changeView = (view: string) => {
    this.setState({
      view,
    });
  };

  renderView = (view: string) => {
    switch (view) {
      case 'make-request':
        return (
          <MakeRequest
            firstLogin={this.props.firstLogin}
            fetchOREA={this.fetchOREA}
            testOREADown={this.testOREADown}
            closeDialog={this.props.closeDialog}
            changeView={this.changeView}
          />
        );
      case 'cannot-reach':
        return <CannotReach changeView={this.changeView} />;
      case 'do-it-later':
        return <DoItLater closeDialog={this.props.closeDialog} />;
      case 'verification-failed':
        return <VerificationFailed changeView={this.changeView} />;
      case 'verification-success':
        return <VerificationSuccess closeDialog={this.props.closeDialog} />;
      default:
        return null;
    }
  };

  render() {
    const { classes } = this.props;
    const { view } = this.state;

    return (
      <div className={classes.dialogContainer}>
        {this.renderView(view)}
      </div>
    );
  }

}

export default compose(
  withStyles(styles),
  withSendOREAGuid,
  withAppUser,
)(OREAValidationC);
