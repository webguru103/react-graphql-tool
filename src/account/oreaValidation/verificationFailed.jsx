import React from 'react';
import { withStyles } from '@material-ui/core/styles';

import Button from '../../shared/button/button';
import styles from './verificationFailed.styles';
import c from './constants';
import IllSmallLock from '../shared/icons/ill_small_lock.svg';

type PropType = {
  classes: Object,
  changeView: Function,
}

export const VerificationFailedC = ({ classes, changeView }: PropType) => (
  <React.Fragment>
    <p className={classes.screenTitle} data-testid="verification-failed">{c.VER_FAILURE_TITLE}</p>
    <div className={classes.notice}>
      <div className={classes.noticeImageContainer}>
        <IllSmallLock style={{ width: '80px' }} />
      </div>
      <div className={classes.noticeText}>
        {c.VER_FAILURE_MSG}
      </div>
    </div>
    <Button
      testId="change-view-make-request"
      classes={{ button: classes.singleButton }}
      text={c.TRY_AGAIN}
      onClick={() => changeView('make-request')}
      fullWidth
    />
  </React.Fragment>
);

export default withStyles(styles)(VerificationFailedC);
