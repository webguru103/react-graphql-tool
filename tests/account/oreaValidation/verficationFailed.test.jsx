/* eslint import/no-extraneous-dependencies: 0 */

import * as React from 'react';
import 'jest-dom/extend-expect';
import { render, cleanup, fireEvent } from 'react-testing-library';
import { withContext } from '../../test-helpers/contextWrapper';
import { VerificationFailedC } from '../../../src/account/oreaValidation/verificationFailed';

describe('Verification Failed Screen', () => {
  afterEach(cleanup);
  const El = withContext(VerificationFailedC);
  it('renders', () => {
    const { container } = render(<El classes={{}} />);
    expect(container).toMatchSnapshot();
  });

  it('calls change view function correctly', () => {
    const changeView = jest.fn();
    const { getByTestId } = render(<El classes={{}} changeView={changeView} />);
    const button = getByTestId('change-view-make-request');

    fireEvent(
      button,
      new MouseEvent('click', { bubbles: true, cancelable: true }),
    );
    expect(changeView).toHaveBeenCalledTimes(1);
    expect(changeView).toBeCalledWith('make-request');
  });
});
