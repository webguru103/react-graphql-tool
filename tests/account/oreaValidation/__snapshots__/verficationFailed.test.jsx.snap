// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verification Failed Screen renders 1`] = `
<div>
  <p
    data-testid="verification-failed"
  >
    <span>
      Verification failed
    </span>
  </p>
  <div>
    <div>
      <svg
        style="width: 80px;"
        viewBox="0 0 110 110"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          transform="translate(17 8)"
        >
          <rect
            fill="#191b24"
            fill-opacity=".05"
            height="57"
            rx="2"
            width="73"
            y="37"
          />
          <path
            d="M5 32.4h70a1 1 0 0 1 1 1V89a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V33.4a1 1 0 0 1 1-1"
            fill="#ebba16"
          />
          <path
            d="M50.8 57.6c0-5.965-4.835-10.8-10.8-10.8-5.965 0-10.8 4.835-10.8 10.8 0 4.741 3.074 8.726 7.326 10.177-.04.209-.126.403-.126.623v7.2c0 1.987 1.613 3.6 3.6 3.6s3.6-1.613 3.6-3.6v-7.2c0-.22-.086-.414-.126-.623C47.726 66.326 50.8 62.341 50.8 57.6"
            fill="#38454f"
          />
          <path
            d="M35 7.2c9.925 0 13.896-7.2 0-7.2S9.8 11.304 9.8 25.2v7.2H17v-7.2c0-9.925 8.07-18 18-18"
            fill="#191b24"
            fill-opacity=".05"
          />
          <path
            d="M22 25.2c0-9.925 8.07-18 18-18 9.925 0 18 8.07 18 18v7.2h7.2v-7.2C65.2 11.304 53.896 0 40 0 26.104 0 14.8 11.304 14.8 25.2v7.2H22v-7.2"
            fill="#424a60"
          />
        </g>
      </svg>
    </div>
    <div>
      <span>
        Sorry, but we were not able to verify your identity with OREA. Please check your information and try again.
      </span>
    </div>
  </div>
  <button
    class="MuiButtonBase-root-31 MuiButton-root-5 MuiButton-contained-16 MuiButton-containedPrimary-17 ButtonC-primary-2 MuiButton-raised-19 MuiButton-raisedPrimary-20 MuiButton-fullWidth-30 ButtonC-button-1"
    data-testid="change-view-make-request"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-6"
    >
      <span>
        Try Again
      </span>
    </span>
  </button>
</div>
`;
